dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -396,8 +415,7 @@ bool sortItemIsInGroupRec(Item* sort_item, Item* group_item)
   found = group_item->eq(sort_item, false);
   if (!found && sort_item->type() == Item::REF_ITEM)
   {
-    Item_ref* ifp_sort_ref = reinterpret_cast<Item_ref*>(sort_item);
-    found = sortItemIsInGroupRec(*ifp_sort_ref->ref, group_item);
+    found = sortItemIsInGroupRec(directRefOrRefLoop(sort_item), group_item);
   }
```

Comment:
Why are we looping here? The original code did not have loop through the `Item`s here. We should follow the same code semantics as in the Spider example where the `real_item()` Item is pulled before any further processing. So the above code should look something like below:
```c++
  if (sort_item->type() == Item::REF_ITEM &&
      ((Item_ref*)sort_item)->ref_type() == Item_ref::DIRECT_REF)
  {
    sort_item= sort_item->real_item();
  }

  if (!found && sort_item->type() == Item::REF_ITEM)
  {
    Item_ref* ifp_sort_ref = reinterpret_cast<Item_ref*>(sort_item);
    found = sortItemIsInGroupRec(*ifp_sort_ref->ref, group_item);
  }
```

Reply:
I disagree. Semantically we need a real element here. directRefOrRefLoop returns either the first real element or the item itself if itsn't a ref.

Reply:
I am open to reconsider this if you elaborate with a counter example.